microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import Path
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import Gpu, SerialConsole
+from lisa.features.gpu import ComputeSDK
+from lisa.operating_system import Redhat
+from lisa.sut_orchestrator.azure.common import AZURE, AzureNodeSchema
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lsmod
+from lisa.util import LisaException, SkippedException
+from lisa.util.perf_timer import create_timer
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+    tags=[],
+)
+class gpu(TestSuite):
+    grid_skus = ["Standard_NV"]
+    cuda_skus = ["Standard_NC", "Standard_ND"]
+
+    def __init__(self, node: Node) -> None:
+        self.gpu_feature = node.features[Gpu]
+        self.serial_console = node.features[SerialConsole]
+
+    def _validate_node_reboot(
+        self, node: Node, case_name: str, case_path: Path
+    ) -> None:
+        try:
+            timer = create_timer()
+            node.reboot()
+            self.log.info(f"node '{node.name}' rebooted in {timer}")
+        except Exception as identifier:
+            # if there is any panic, fail before partial pass
+            self.serial_console.check_panic(saved_path=case_path, stage="reboot")
+
+            # if node cannot be connected after reboot, it should be failed.
+            if isinstance(identifier, LisaException) and str(identifier).startswith(
+                "cannot connect to TCP port"
+            ):
+                raise LisaException(f"after reboot, {identifier}")
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
```

Comment:
The metadata should specify how to verify.

Reply:
Can please you also run spell checking on your commits?